docarray/array/mixins/io/json.py

```diff
@@ -51,30 +49,27 @@ def load_json(
 
         :return: a DocumentArrayLike object
         """
-
-        from .... import Document
-
-        constructor = Document.from_json
         if hasattr(file, 'read'):
             file_ctx = nullcontext(file)
-        elif os.path.exists(file):
-            file_ctx = open(file, 'r', encoding=encoding)
         else:
-            file_ctx = nullcontext(json.loads(file))
-            constructor = Document.from_dict
+            file_ctx = open(file, 'r', encoding=encoding)
 
         with file_ctx as fp:
-            return cls([constructor(v, protocol=protocol) for v in fp], **kwargs)
+            return cls.from_json(fp.read(), protocol=protocol, **kwargs)
 
     @classmethod
     def from_json(
         cls: Type['T'],
         file: Union[str, bytes, bytearray, TextIO],
         protocol: str = 'jsonschema',
-        encoding: str = 'utf-8',
         **kwargs
     ) -> 'T':
-        return cls.load_json(file, protocol=protocol, encoding=encoding, **kwargs)
+        from .... import Document
+
+        json_docs = json.loads(file)
```

Comment:
are your sure this maps to the type hint? 

```
        file: Union[str, bytes, bytearray, TextIO],

```

either this line or type hint, one of them is wrong

Reply:
fixed. Only str, bytes and bytearray are kept while TextIO and str are supported for load_json